src/os/bluestore/KernelDevice.cc

```diff
@@ -342,6 +342,7 @@ void KernelDevice::_aio_thread()
 					 aio, max);
     if (r < 0) {
       derr << __func__ << " got " << cpp_strerror(r) << dendl;
+      assert(0 == "got unexpected error from io_getevents");
```

Comment:
@liewegas do you mean I should ignore EINTR here? If yes, I don't follow...

In bluestore/aio.cc, there is:

int aio_queue_t::get_next_completed(int timeout_ms, aio_t **paio, int max)
{
  io_event event[max];
  struct timespec t = {
    timeout_ms / 1000,
    (timeout_ms % 1000) * 1000 * 1000
  };

 ```
 int r = 0;
  do {
    r = io_getevents(ctx, 1, max, event, &t);
  } while (r == -EINTR);
```

So io_getevents will be called again if r == -EINTR, so I don't think the return value(r) could be -EINTR. Please correct me if I have anything wrong.